torch/csrc/distributed/autograd/functions/recvrpc_backward.cpp

```diff
@@ -30,21 +30,23 @@ variable_list RecvRpcBackward::apply(variable_list&& grads) {
     }
   }
 
+  auto sharedContext = autogradContext_.lock();
+  TORCH_CHECK(sharedContext, "Autograd context no longer valid");
```

Comment:
nit: TORCH_INTERNAL_ASSERT ?

Reply:
This isn't really a bug if this check fails. It is possible that this check fails because distributed autograd failed and another thread cleaned up the autograd context. In this case, its fine for this line to throw an exception to reporting an error upstream.

Reply:
>  its fine for this line to throw an exception to reporting an error upstream

Not sure to follow this sentence?

If this is a user facing error, it should be a bit more explicit no?
The internal assert will tell the user to report a bug to pytorch. But a regular one will just say that. Is the idea that the user should look at the other processes for the real error?

Reply:
Yes, the idea is that the user should look at other processes for the real error. You have a good point, I'll make this error more explicit.